diff -a -u -r1.6 ControlPanel.py
--- frameworkd/ossimframework/ControlPanel.py	28 Mar 2005 13:38:15 -0000	1.6
+++ frameworkd/ossimframework/ControlPanel.py	12 Apr 2005 15:12:24 -0000
@@ -160,24 +160,8 @@
 
                 threshold = self.__conf["threshold"]
 
-
-                # get last value for day level
-                rrd_info = self.__get_rrd_value("N-15m", "N", rrd_file)
-
-                query = """
-            UPDATE control_panel
-                SET c_sec_level = %f, a_sec_level = %f
-                WHERE id = 'global_%s' and time_range = 'day'
-            """ % (rrd_info["max_c"], rrd_info["max_a"], user)
-
-                print "updating level (day):  C=%s%%, A=%s%%" % \
-                    (rrd_info["max_c"], rrd_info["max_a"])
-
-                self.__conn.exec_query(query)
-
-
-                # calculate average for week, month and year levels
-                for range in ["week", "month", "year"]:
+                # calculate average for day, week, month and year levels
+                for range in ["day", "week", "month", "year"]:
 
                     range2date = {
                         "day"  : "N-1D",
